### PR TITLE
Generalize ubuntu-headnode example

### PR DESCRIPTION
**Pull request by zenhack**
It now can generate CentOS images as well, and it should be able to add
more.

We want CentOS headnodes for the NEU gear we're setting up.

### CODE REVIEW

**Comment by gsilvis** [comment]
This looks good to me.  +1

File: INSTALL.rst [hunk 1]
```diff
@@ -173,9 +173,9 @@ activate on boot, with::
...
 
-The scripts in ``examples/ubuntu-headnode`` can be used to build an ubuntu
-14.04 disk image with a default root password. Read the README in that
-directory for more information.
+The scripts in ``examples/cloud-img-with-passwd`` can be used to build
+an ubuntu 14.04 or centos 7 disk image with a default root password. Read
+the README in that directory for more information.
 
 Once the disk image is built, copy ito the storage pool directory (here we
 assume it is caled ``base.img``)::
```
Comment (lokI8) [inline INSTALL.rst]: "called"?

File: examples/cloud-img-with-passwd/Makefile [hunk 2]
```diff
@@ -12,14 +12,13 @@
...
 # governing permissions and limitations under the License.
 
-all: build
+include $(distro).mk
 
-MIRROR := http://cloud-images.ubuntu.com/trusty/current
-IMG_IN := trusty-server-cloudimg-amd64-disk1.img
+all: build
 $(IMG_IN):
 	wget $(MIRROR)/$(IMG_IN)
```
Comment (lokI8) [inline examples/cloud-img-with-passwd/Makefile]: Where does it get the mirror?

File: INSTALL.rst (hunk 1 above)
Comment (zenhack) [inline INSTALL.rst]: Already there apparently, but I'll fix it while I'm here.

**Comment by zenhack** [comment]
@lokI8, I've addressed your comments, can you make another pass?

**Comment by lokI8** [comment]
Looks good, +1. Sorry this took so long.

**Pull request merged**